### PR TITLE
Cu 86b17fr3j responsive styling for navigation bar ethan freire

### PR DESCRIPTION
**Pull request by ethanfreire**

### CODE REVIEW

**Comment by allantito** [comment]
Task linked: [CU-86b17fr3j Responsive styling for navigation bar](https://app.clickup.com/t/86b17fr3j)

File: P1/Project_Code/frontend/src/services/useViewport.jsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { useState, useEffect } from 'react';
+
+const useViewport = () => {
```
Comment (ethanfreire) [inline P1/Project_Code/frontend/src/services/useViewport.jsx:3]: Where is the useViewport being used?

File: P1/Project_Code/frontend/src/Theme.styles.jsx [hunk 2]
```diff
@@ -4,13 +4,14 @@ import { createTheme } from "@mui/material";
...
 const theme = createTheme({
     // breakpoints: {
-// 	values: {
-// 		xs: 0,
-// 		sm: 375,
-// 		md: 786,
-// 		lg: 1440,
-// 		xl: 1536,
-// 	},
+	// values: {
```
Comment (ethanfreire) [inline P1/Project_Code/frontend/src/Theme.styles.jsx:7]: Are these comment neccessary? Are these breakpoint values being referenced in the application?

File: P1/Project_Code/frontend/src/components/Layout/Layout.jsx [hunk 3]
```diff
@@ -25,16 +26,10 @@ export const Layout = () => {
...
-        container
-        justifyContent="center"
-        style={{ paddingTop: "1rem" }}
-      >
+      <NavigationBarGridContainer item xs={0} md={0} lg={1} container justifyContent="center">
         <NavigationBar></NavigationBar>
-      </Grid>
-      <Grid item xs={11}>
+      </NavigationBarGridContainer>
+      <Grid item xs={12} md={12} lg={11}>
```
Comment (ryanRATM) [inline P1/Project_Code/frontend/src/components/Layout/Layout.jsx:32]: you should be able to remove the md={12}.  MaterialUI adopts/inherits the value defined at a lower screen size (xs={12})

File: P1/Project_Code/frontend/src/components/Navigation/NavigationBar.jsx [hunk 4]
```diff
@@ -15,13 +15,18 @@ import {
...
+import useViewport from "../../services/useViewport.jsx";
 
 export const NavigationBar = () => {
   const location = useLocation();
   const [activePath, setActivePath] = useState(location.pathname);
-
+  const { width } = useViewport();
   const { user, logoutUser } = useContext(UserContext);
 
+  const isMobile = width <= 767;
```
Comment (ryanRATM) [inline P1/Project_Code/frontend/src/components/Navigation/NavigationBar.jsx:26]: https://mui.com/material-ui/customization/breakpoints/
https://mui.com/material-ui/react-use-media-query/
look at how to do screen checks this way

Comment (ethanfreire) [reply P1/Project_Code/frontend/src/components/Navigation/NavigationBar.jsx:26]: Agreed.I'm creating a backlog ticket to refactor this code to use useMediaQuery hook.

File: P1/Project_Code/frontend/src/components/Layout/Layout.jsx (hunk 3 above)
Comment (ethanfreire) [reply P1/Project_Code/frontend/src/components/Layout/Layout.jsx:32]: Agreed, this code should be simplified.

**Pull request merged**